heartbeat/monitors/util.go

```diff
@@ -13,20 +12,39 @@ import (
 )
 
 type funcJob struct {
-	name, typ string
+	settings JobSettings
 	funcTask
 }
 
 type funcTask struct {
 	run func() (common.MapStr, []TaskRunner, error)
 }
 
+// IPSettings provides common configuration settings for IP resolution and ping
+// mode.
 type IPSettings struct {
 	IPv4 bool     `config:"ipv4"`
 	IPv6 bool     `config:"ipv6"`
 	Mode PingMode `config:"mode"`
 }
 
+// JobSettings configures a Job name and global fields to be added to every
+// event.
+type JobSettings struct {
+	Name   string
+	Fields common.MapStr
+}
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported type HostJobSettings should have comment or be unexported